### PR TITLE
🐛 upgrade golangci-lint for go 1.18 support

### PR DESCRIPTION
**Pull request by apricote**
**What this PR does / why we need it**:

With versions lower than 1.45.0, golangci-lint returns a lot of invalid errors when running in Go 1.18+. Because of this, all PR pipeline jobs `pull-cluster-api-provider-openstack-test` fail.

This commit upgrades golangci-lint to the latest released version and fixes all linting errors that came up.

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

/hold

### CODE REVIEW

**Comment by tobiasgiese** [comment]
/lgtm
/approve
/hold cancel

thanks @apricote 👌🏻

File: pkg/record/recorder.go [hunk 1]
```diff
@@ -45,20 +46,20 @@ func InitFromRecorder(recorder record.EventRecorder) {
 
 // Event constructs an event from the given information and puts it in the queue for sending.
 func Event(object runtime.Object, reason, message string) {
-	defaultRecorder.Event(object, corev1.EventTypeNormal, strings.Title(reason), message)
+	defaultRecorder.Event(object, corev1.EventTypeNormal, cases.Title(language.Und).String(reason), message)
```
Comment (seanschneeweiss) [inline pkg/record/recorder.go:49]: Out of curiosity, what is the `language.Und` doing?

**Review by seanschneeweiss — COMMENTED** [review]
/lgtm
Thank you very much for fixing this.

**Comment by tobiasgiese** [comment]
/check-cla
/retest
/approve
/lgtm

I think prow had some issues today

**Comment by jichenjc** [comment]
@apricote guess you forgot to run `go mod tidy` to update go.sum?

File: api/v1alpha3/openstackcluster_types.go [hunk 2]
```diff
@@ -30,7 +30,6 @@ const (
 
 // OpenStackClusterSpec defines the desired state of OpenStackCluster.
 type OpenStackClusterSpec struct {
-
```
Comment (jichenjc) [inline api/v1alpha3/openstackcluster_types.go:33]: unrelated change..

File: pkg/record/recorder.go (hunk 1 above)
Comment (jichenjc) [reply pkg/record/recorder.go:49]: same question here..

File: api/v1alpha3/openstackcluster_types.go (hunk 2 above)
Comment (apricote) [reply api/v1alpha3/openstackcluster_types.go:33]: These are changes that `golangci-lint` requested after I upgrade it. In this case the structs were not formatted with gofumpt

File: pkg/record/recorder.go (hunk 1 above)
Comment (apricote) [reply pkg/record/recorder.go:49]: I think this is for `Undefined` language, it's not really explained in the [docs]. Using this language tag returned the expected result, as you can see in this Go Playground example: https://go.dev/play/p/fNew-KrewPY

**Comment by apricote** [comment]
> @apricote guess you forgot to run `go mod tidy` to update go.sum?

Looks like I forgot to run it a second time after fixing the linting issues. New commit is up

File: pkg/record/recorder.go (hunk 1 above)
Comment (lentzi90) [reply pkg/record/recorder.go:49]: Makes sense. Maybe change it to `language.English` then?

Comment (apricote) [reply pkg/record/recorder.go]: I think that makes sense, considering the reasons should be in english so all consumers of them can understand them.

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by pierreprinetti** [comment]
pass! hurray!

**Comment by apricote** [comment]
/cherry-pick release-0.5

third time's a charm

**Comment by apricote** [comment]
/cherry-pick release-0.5

Third time also didn't work and I forgot about this afterwards, we are currently hitting this issue on #1204.
